components/nvs/example/main/nvs_example.cpp

```diff
@@ -12,80 +12,145 @@ extern "C" void app_main(void) {
+    std::string str;
+    nvs.get_or_set_var("system", "string", str, std::string("default"), ec);
+    if (ec) {
+      fmt::print("Error: {}\n", ec.message());
+    } else {
+      fmt::print("String = {}\n", str);
+    }
+    ec.clear();
+
+    // test setting a string value
+    str = "hello";
+    nvs.set_var("system", "string", str, ec);
+    if (ec) {
+      fmt::print("Error: {}\n", ec.message());
+    } else {
+      fmt::print("String set to '{}'\n", str);
+    }
     ec.clear();
+
+    counter++;
+
+    if (counter > 10) {
+      nvs.erase(ec);
+      nvs.init(ec);
+      counter = 0;
+      fmt::print("NVS erased, Reset Counter set to 0\n");
+      ec.clear();
+    }
+
+    nvs.set_var("system", "reset_counter", counter, ec);
+    fmt::print("Next Reset Counter will be = {}\n", counter);
+    fmt::print("NVS example complete!\n");
+    //! [nvs example]
   }
 
-  nvs.set_var("system", "reset_counter", counter, ec);
-  fmt::print("Next Reset Counter will be = {}\n", counter);
-  fmt::print("NVS example complete!\n");
-  //! [nvs example]
+  {
+    //! [nvshandle example]
+    // Init NVS
+    std::error_code ec;
+    espp::Nvs nvs;
+    nvs.init(ec);
+    ec.clear();
+    
+    // Open
+    printf("\n");
+    printf("Opening Non-Volatile Storage (NVS) handle... ");
+    // Handle will automatically close when going out of scope or when it's reset.
+    espp::NVSHandle storage("storage", ec);
+    printf("Done\n");
+    ec.clear();
+    
+    // Read
+    printf("Reading restart counter from NVS ... ");
+    int32_t restart_counter = 0; // value will default to 0, if not set yet in NVS
+    storage.get("restart_counter", restart_counter, ec);
+    if (ec) {
+      if (ec.value() == static_cast<int>(NvsErrc::Key_Not_Found)) {
+        printf("The value is not initialized yet!\n");
+      } else if (ec.value() == static_cast<int>(NvsErrc::Read_NVS_Failed)) {
+        printf("Failed to read from NVS: %s\n", ec.message().c_str());
+      } else {
+        printf("An error occurred: %s\n", ec.message().c_str());
+      }
+    } else {
+      printf("Done\n");
+      printf("Restart counter = %" PRIu32 "\n", restart_counter);
+    }
+    ec.clear();
+
+    // Write
+    printf("Updating restart counter in NVS ... ");
+    restart_counter++;
+    storage.set("restart_counter", restart_counter, ec);
+    printf("Done\n");
+    ec.clear();
+
+    // Commit written value.
+    // After setting any values, nvs_commit() must be called to ensure changes are written
+    // to flash storage. Implementations may write to storage at other times,
+    // but this is not guaranteed.
+    printf("Committing updates in NVS ... ");
+    storage.commit(ec);
+    printf("Done\n");
+    ec.clear();
+
+    printf("\n");
+
+    // Restart module
+    for (int i = 10; i >= 0; i--) {
+        printf("Restarting in %d seconds...\n", i);
+        vTaskDelay(1000 / portTICK_PERIOD_MS);
+    }
+    printf("Restarting now.\n");
+    fflush(stdout);
+    esp_restart();
```

Comment:
I believe this code is lifted from esp-idf example, though I think it's better to not have it automatically restart, given that the serial connection provides restart mechanisms. This way you have more than 10 seconds to read the output and validate it's correct :)